### PR TITLE
Add API endpoints

### PR DESCRIPTION
**Pull request by NurkaAmre**
### In this PR we implemented following features:

- Created API endpoint to list all posts for a user.
- Created  API endpoint to list all comments for a user's post.
- Created API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.
WhatsApp Image 2023-02-10 at 17 16 14
WhatsApp Image 2023-02-10 at 17 15 24
WhatsApp Image 2023-02-10 at 17 15 54

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def create
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: @comment.errors, status: :unprocessable_entity
+    end
+  end
```
Comment (ichala) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] As mentioned in the requirements : 

image

you should user current user instead 🙏

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:author_id, :post_id, :text)
+  end
```
Comment (ichala) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] kindly fix this error when creating a new comment 🙏 

image

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- EndPoints Created  ✔️ 


 ## Required Changes ♻️

- [ ] **_🛑  Check The Comments Below_**




## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/comments_controller.rb (hunk 1 above)
Comment (NurkaAmre) [reply app/controllers/api/v1/comments_controller.rb]: Its not clear what U mean @ichala can you give more hints?

Comment (ichala) [reply app/controllers/api/v1/comments_controller.rb]: We Talked on slack 💯

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello TEAM! 👋 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**